Register{Schema,BackendSelect}.cpp: cleanup header includes

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #69840
* __->__ #70021
* #68693

`RegisterSchema.cpp` only uses strings to register operator schemas,
so doesn't need to include any operator headers at all (except
indirectly through `torch/library.h`).

`RegisterBackendSelect.cpp` only needs the dispatcher API.

Differential Revision: [D33160028](https://our.internmc.facebook.com/intern/diff/D33160028)

peterbell10 committed:
Register{Schema,BackendSelect}.cpp: cleanup header includes

`RegisterSchema.cpp` only uses strings to register operator schemas,
so doesn't need to include any operator headers at all (except
indirectly through `torch/library.h`).

`RegisterBackendSelect.cpp` only needs the dispatcher API.

[ghstack-poisoned]

peterbell10 committed:
Rebase to fix merge conflicts on "Register{Schema,BackendSelect}.cpp: cleanup header includes"


`RegisterSchema.cpp` only uses strings to register operator schemas,
so doesn't need to include any operator headers at all (except
indirectly through `torch/library.h`).

`RegisterBackendSelect.cpp` only needs the dispatcher API.

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33160028).

peterbell10 committed:
Update on "Register{Schema,BackendSelect}.cpp: cleanup header includes"


`RegisterSchema.cpp` only uses strings to register operator schemas,
so doesn't need to include any operator headers at all (except
indirectly through `torch/library.h`).

`RegisterBackendSelect.cpp` only needs the dispatcher API.

Differential Revision: [D33160028](https://our.internmc.facebook.com/intern/diff/D33160028)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33160028).

peterbell10 committed:
Update on "Register{Schema,BackendSelect}.cpp: cleanup header includes"


`RegisterSchema.cpp` only uses strings to register operator schemas,
so doesn't need to include any operator headers at all (except
indirectly through `torch/library.h`).

`RegisterBackendSelect.cpp` only needs the dispatcher API.

Differential Revision: [D33160028](https://our.internmc.facebook.com/intern/diff/D33160028)

[ghstack-poisoned]

peterbell10 committed:
Update on "Register{Schema,BackendSelect}.cpp: cleanup header includes"


`RegisterSchema.cpp` only uses strings to register operator schemas,
so doesn't need to include any operator headers at all (except
indirectly through `torch/library.h`).

`RegisterBackendSelect.cpp` only needs the dispatcher API.

Differential Revision: [D33160028](https://our.internmc.facebook.com/intern/diff/D33160028)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33160028).

peterbell10 committed:
Update on "Register{Schema,BackendSelect}.cpp: cleanup header includes"


`RegisterSchema.cpp` only uses strings to register operator schemas,
so doesn't need to include any operator headers at all (except
indirectly through `torch/library.h`).

`RegisterBackendSelect.cpp` only needs the dispatcher API.

Differential Revision: [D33160028](https://our.internmc.facebook.com/intern/diff/D33160028)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33160028).

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33160028).

peterbell10 committed:
Update on "Register{Schema,BackendSelect}.cpp: cleanup header includes"


`RegisterSchema.cpp` only uses strings to register operator schemas,
so doesn't need to include any operator headers at all (except
indirectly through `torch/library.h`).

`RegisterBackendSelect.cpp` only needs the dispatcher API.

Differential Revision: [D33160028](https://our.internmc.facebook.com/intern/diff/D33160028)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33160028).

peterbell10 committed:
Rebase and fix merge conflicts on "Register{Schema,BackendSelect}.cpp: cleanup header includes"


`RegisterSchema.cpp` only uses strings to register operator schemas,
so doesn't need to include any operator headers at all (except
indirectly through `torch/library.h`).

`RegisterBackendSelect.cpp` only needs the dispatcher API.

Differential Revision: [D33160028](https://our.internmc.facebook.com/intern/diff/D33160028)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33160028).

peterbell10 committed:
Update on "Register{Schema,BackendSelect}.cpp: cleanup header includes"


`RegisterSchema.cpp` only uses strings to register operator schemas,
so doesn't need to include any operator headers at all (except
indirectly through `torch/library.h`).

`RegisterBackendSelect.cpp` only needs the dispatcher API.

Differential Revision: [D33160028](https://our.internmc.facebook.com/intern/diff/D33160028)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33160028).

peterbell10 committed:
Update on "Register{Schema,BackendSelect}.cpp: cleanup header includes"


`RegisterSchema.cpp` only uses strings to register operator schemas,
so doesn't need to include any operator headers at all (except
indirectly through `torch/library.h`).

`RegisterBackendSelect.cpp` only needs the dispatcher API.

Differential Revision: [D33160028](https://our.internmc.facebook.com/intern/diff/D33160028)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33160028).

peterbell10 committed:
Update on "Register{Schema,BackendSelect}.cpp: cleanup header includes"


`RegisterSchema.cpp` only uses strings to register operator schemas,
so doesn't need to include any operator headers at all (except
indirectly through `torch/library.h`).

`RegisterBackendSelect.cpp` only needs the dispatcher API.

Differential Revision: [D33160028](https://our.internmc.facebook.com/intern/diff/D33160028)

[ghstack-poisoned]
